src/commontransformations.jl

```diff
@@ -310,56 +317,89 @@ Base.show(io::IO, r::RotationXY) = print(io, "RotationXY($(r.angle))")
 Base.show(io::IO, r::RotationYZ) = print(io, "RotationYZ($(r.angle))")
 Base.show(io::IO, r::RotationZX) = print(io, "RotationZX($(r.angle))")
 
+# It's a little fiddly to support all possible point container types, but this should do a good majority of them!
+transform(trans::RotationXY, x::Vector) =    [x[1]*trans.cos - x[2]*trans.sin, x[1]*trans.sin + x[2]*trans.cos, x[3]]
```

Comment:
`AbstractVector` gets caught be the bottom-most function. If the user has implemented matrix multiplication on their vector, they will get the correct `similar` type.